internal/prompt/prompt.go

```diff
@@ -16,10 +16,28 @@ package prompt
 
 import (
 	"fmt"
+	"io"
+	"os"
 
 	"github.com/AlecAivazis/survey/v2"
+	"github.com/mattn/go-isatty"
 )
 
+// IsTerminalInput returns true if input device is a terminal
+// Will return false in the case of input being piped.
+func IsTerminalInput() bool {
+	return IsTerminalInputStream(os.Stdin)
+}
+
+// isTerminalInput returns true if input device is a terminal
+// Will return false in the case of the reader not being a terminal.
+func IsTerminalInputStream(stream io.Reader) bool {
+	if inputStream, ok := stream.(*os.File); ok {
+		return isatty.IsTerminal(inputStream.Fd()) || isatty.IsCygwinTerminal(inputStream.Fd())
+	}
+	return false
+}
```

Comment:
[nit] good on the idea but I would avoid the global `IsTerminalInput` given you always call this in the context of a cobra command, then on the usage of `IsTerminalInputStream(stream io.Reader)` you always call it with `os.Stdin` this may introduce some edge-cases given the cobra context, so to make this correct you should call IsTerminal always with `cmd.InOrStdin()`

I think I mentioned this before but I believe the ideal approach would be something similar to OutputOpts https://github.com/mongodb/mongocli/blob/9c641961e016b06040b06c0ccff9adbb3a4e701a/internal/cli/output_opts.go#L41-L45

Reply:
Ah right, I didn't quite get what you were suggesting. I've added an `InputOpts` struct and call it where I think is appropriate for its use - I've also removed the functions from `prompt.go` as they're redundant now. Is that what you were looking for?

I've also reverted my changes to `internal/cli/atlas/dbusers/update.go` as I didn't understand that not specifying a password is a valid use of the command, so the regular `Prompt` work flow wouldn't work.